### DISCUSSION TITLE
Updating ESP07 with limited internal memory (180KB) via OTA

### DISCUSSION THREAD

**Discussion by evenblan** [arendst/Tasmota#19406] [Q&A]
Hello Tasmota community,

I am currently running Tasmota 11 on an ESP07 chip which is located in a hard-to-reach place. I'd like to perform an OTA update on this device, but I've encountered a challenge. The internal memory of the chip is limited to only 180KB, and from my understanding, the minimal firmware is around 240KB.

Given the constraints:

The device is not easily accessible for direct interfacing.
The limited internal memory.
Is there any recommended way or workaround to successfully perform an OTA update without risking to brick the device?

Additionally, I don't have a use for LittleFS in my current application with this ESP07. Is there a way to remove or replace the LittleFS to potentially save space and facilitate the OTA process?

Furthermore, I was wondering if it's feasible to compile the firmware with even fewer features to fit within the 180KB limit. For instance, is it possible to remove specific commands, MQTT functionality, or even the entire web interface to conserve space?

I would greatly appreciate any guidance or insights you might provide on these matters.

![image](https://github.com/arendst/Tasmota/assets/63598896/10205b30-2945-4c8b-a445-581dfe45cd90)

Thank you in advance for your assistance and time!

Best regards,

**Comment by sfromis** [comment]
If you have made a special build to squeeze a file system within the 1MB of flash, you have cut off normal ways of doing OTA upgrade of firmware. While you *can* remove features, including web UI, but not specific commands, this would not get you near the 180 KB.

Looks like you may be able to do a firmware replacement OTA (but not upgrade existing config) using:
https://github.com/tasmota/ESP8266MinimalHTTPUpdater

If you want to stay with the ESP-07 form factor, I'd recommend getting modules with 4MB of flash.
